enhancements/windows-containers/bring-your-own-windows-host.md

```diff
@@ -0,0 +1,182 @@
+* [Handle upgrades](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+  of all components installed by WMCO.
+
+### Non-Goals
+
+* Installing the container runtime in the Windows instance
+* Managing the Windows operating system(OS) including handling of OS updates
+* OpenShift Builds
+* OpenShift DeploymentConfigs
+
+## Proposal
+
+The cluster admin will create a ConfigMap called _windows-instances_ in the
+_openshift-windows-machine-config-operator_ namespace. WMCO will be configured
+to watch for this special ConfigMap, whose data section will have the following
+layout:
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: windows-instances
+  namespace: openshift-windows-machine-config-operator
+data:
+  instance.1: |-
+    username=Administrator
+    ip-address=10.1.42.1
+  instance.2: |-
+    username=core
+    ip-address=10.1.42.2
+```
+WMCO will configure or de-configure instances based on the contents of the data
+section and will use the [cloud-private-key]((https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator.md#design-details)
+to communicate with them.
+
+### User Stories
+
+#### Story 1
+As an OpenShift cluster administrator I would like to add an existing Windows
+instance to the cluster as a worker node.
+
+#### Story 2
+As an OpenShift cluster administrator I would like to remove an existing Windows
+instance from the cluster as a worker node.
+
+#### Story 3
+As an OpenShift cluster administrator I would like to have the Kubernetes
+components on Windows nodes upgraded when new Kubernetes versions or patch
+versions are released.
+
+### Risks and Mitigations
+
+The same risk and mitigation that apply to Windows nodes added using MachineSets
+to an IPI cluster applies here. In addition there is potential for a for a node
+to become unstable if configuration fails and will require administrator
+intervention to fix. There is no real mitigation here other that for us to test
+this feature thoroughly in CI.
+
+## Design Details
+
+WMCO will include a ConfigMap controller that watches for changes to the
+_windows-instances_ ConfigMap. Any change will cause WMCO to compare the list
+of instances with the set of existing Windows nodes. The comparison will be
+based on the instance and node's IP addresses. WMCO will then react in the
+following manner:
+* If a new entry is found i.e. the list of nodes does not contain one that map
+  to the instances' IP address, the instance will be configured by WMCO. The
+  configuration process will be identical to configuring instances added using
+  a MachineSet in an IPI cluster except in addition, WMCO will also have to
+  handle CSR approvals.
+* If the list of nodes has IP addresses that do not map to any in the list of
+  instances, WMCO will perform an uninstall operation on the node. This will
+  involve:
```

Comment:
do we also need to do anything for the networking bits setup? Just a question for my understanding.

Reply:
Yes  that's what I was thinking about, we haven't mentioned anything about networking in the proposal hence I was asking if the setup would be same as the IPI clusters.